charts/aws-efs-csi-driver/templates/controller-deployment.yaml

```diff
@@ -32,29 +32,29 @@ spec:
       {{- end }}
       nodeSelector:
         kubernetes.io/os: linux
-        {{- with .Values.nodeSelector }}
+        {{- with .Values.controller.nodeSelector }}
         {{- . | toYaml | nindent 8 }}
         {{- end }}
       serviceAccountName: {{ include "aws-efs-csi-driver.serviceAccountName" . }}
       priorityClassName: system-cluster-critical
```

Comment:
It might be worth considering adding the ability to set topologySpreadConstraints while this change is being made.  This is untested but might look something like this.

```
      {{- with .Values.controller.topologySpreadConstraints }}
      {{- $tscLabelSelector := dict "labelSelector" (dict "matchLabels" (dict "app" "efs-csi-controller")) }}
      topologySpreadConstraints:
        {{- range . }}
        - {{ mergeOverwrite . $tscLabelSelector | toJson }}
        {{- end }}
      {{- end }}
```

Reply:
I am not sure of the implications of changing anything here, are we supporting only Kubernetes running on Linux hosts?

Reply:
Not adding `topologySpreadConstraints` for now, sticking with the current `nodeSelector` and `priorityClassName`.

Reply:
works for me, I don't think topologySpreadConstraints would make sense on node, we can always add later to controller if somebody asks for it.